app/controllers/api/v1/friends_controller.rb

```diff
@@ -1,5 +1,7 @@
 class Api::V1::FriendsController < ApplicationController
   require 'debug'
+
+  before_action :current_friend, only:[:update]
```

Comment:
This is a pretty clever use of the before_action hook.

This abstraction is not worth the savings, though.